### PR TITLE
[Refactor] Add PersistentIndex tablet loader for both shared data and shared nothing

### PR DESCRIPTION
**Pull request by luohaha**
Why I'm doing:
In previous code, load persistent index from shared data and shared nothing are using same code copying. 
1. It's very ugly and 
2. It will cause bugs when there is a modification to the persistent index and doesn't modify both parts of the code at the same time.

What I'm doing:
Try to do better, merge two piece of code into one code.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by sevev** [comment]
This PR split the `persistent_index_loader` and `lake_persistent_index_loader`, so modifying the code for `pindex` will still require changes in both places?

File: be/src/storage/persistent_index_tablet_loader.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+public:
+    PersistentIndexTabletLoader(Tablet* tablet) : _tablet(tablet) {}
+    ~PersistentIndexTabletLoader() = default;
+    starrocks::Schema generate_pkey_schema() override;
+    DataDir* data_dir() override;
+    TTabletId tablet_id() override;
+    // return latest applied (publish in cloud native) version
+    StatusOr<EditVersion> applied_version() override;
+    // Do some special setting if need
+    void setting() override{};
```
Comment (wyb) [inline be/src/storage/persistent_index_tablet_loader.h:40]:
```suggestion
    void setting() override {};
```

**Review by wyb — DISMISSED** [review]

File: be/src/storage/persistent_index_tablet_loader.h (hunk 1 above)
Comment (luohaha) [reply be/src/storage/persistent_index_tablet_loader.h:40]: We can't do that, it is clang format required.

**Comment by luohaha** [comment]
> This PR split the `persistent_index_loader` and `lake_persistent_index_loader`, so modifying the code for `pindex` will still require changes in both places?

No, just need

**Review by wyb — APPROVED** [review]

**Review by sevev — APPROVED** [review]

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**